### PR TITLE
New ERC20Upgradeable token with support for migration from legacy MadToken

### PR DESCRIPTION
**Pull request by gusjavaz**
What is changing with this PR?
Added new  ERC20Upgradeable token with support for migration from legacy MadToken

Why are we doing this?
To have an new upgradeable token contract

### CODE REVIEW

**Comment by gusjavaz** [comment]
Implemented reviewer suggestions.
Reopening

File: bridge/contracts/AToken.sol [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        IERC20(_legacyToken).transferFrom(msg.sender, address(this), amount);
+        _mint(msg.sender, amount);
+    }
+
+    function externalMint(address to, uint256 amount) public onlyATokenMinter {
+        _mint(to, amount);
+    }
+
+    function externalBurn(address from, uint256 amount) public onlyATokenBurner {
+        //add require
```
Comment (vtleonardo) [inline bridge/contracts/AToken.sol:44]: Did a double check in the ERC20 contract, the _burn already have the require that we talked about (if the amount was not greater then the guy's balance)

File: bridge/test/token/aToken/burning.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  beforeEach(async function () {
+    fixture = await getFixture();
+    [admin, user] = await ethers.getSigners();
+    await init(fixture);
+    expectedState = await getState(fixture);
+  });
+
+  describe("Testing burning operation", async () => {
+    describe.skip("Business methods with onlyFactory modifier", async () => {
```
Comment (vtleonardo) [inline bridge/test/token/aToken/burning.ts:26]: Is this being skipped?

Comment (gusjavaz) [reply bridge/test/token/aToken/burning.ts]: good spot! Fixed in latest commit

File: bridge/contracts/AToken.sol (hunk 1 above)
Comment (gusjavaz) [reply bridge/contracts/AToken.sol]: Yes I've checked myself too.  Comment removed.

**Pull request merged**